### ISSUE TITLE
Empty `{}` in string interpolation should fail

### ISSUE THREAD

**Issue by JohnnyMorganz** [Kampfkarren/full-moon#261]
#253 
```lua
local _ = `asdf{}`
```
This currently passes, but it should fail.
Luau errors with "SyntaxError: Malformed interpolated string, expected expression inside '{}"

**Comment by jeparlefrancais** [comment]
I don't know if it's preferable to create another issue but I also noticed that the string ``{ {} }`` does not parse in full-moon in the latest version:

```
unexpected double brace for interpolated string. try \\{ if you meant to escape
```

But if you give it a try on https://luau-lang.org/demo, it does work:

![image](https://user-images.githubusercontent.com/35781636/236925887-3365cf02-fa84-46b8-985a-f802b1d4e3c0.png)

**Comment by Kampfkarren** [comment]
That actually probably shouldn't work in Luau either... :P 

But c'est la vie

**Comment by jeparlefrancais** [comment]
Yeah I don't know where I sit with if it should or shouldn't, but I do believe it's a little confusing to work with because it's very hard to predict this:

```lua
local _ = `{{ }}` -- err
local _ = `{ {}}` -- ok
local _ = `{--[]{}}` -- ok
local _ = `\{{` -- ok
local _ = `{{hello}}` -- err
local _ = `{ {hello}}` -- ok
local _ = `\{{hello}}` -- ok
```

😕
